source/MaterialXCore/Interface.cpp

```diff
@@ -116,37 +128,29 @@ OutputPtr PortElement::getConnectedOutput() const
     else if (hasNodeName())
     {
         const string& nodeName = getNodeName();
-        ConstElementPtr startingElement = getParent();
-        if (startingElement)
+        NodePtr node = resolveRootNameReference<Node>(nodeName, root);
+        if (!node)
         {
-            // Look for a node reference above the nodegraph if input is a direct child.
-            if (startingElement->isA<NodeGraph>())
-            {
-                startingElement = startingElement->getParent();
-            }
-            if (startingElement)
+            node = resolveRootNameReference<Node>(nodeName);
+        }
+        if (node)
+        {
+            std::vector<OutputPtr> outputs = node->getOutputs();
```

Comment:
This logic has not changed. As an explicit output is requested and if the output is implicit it will not be found.